### PR TITLE
include Address in CSV export of member info

### PR DESCRIPTION
**Pull request by weedySeaDragon**
PT Story:  https://www.pivotaltracker.com/story/show/145067071

### Changes proposed in this pull request:
1.  added the **AddressExporter** class who has the responsibility for creating the correct 'CSV export' string for either an `Address` or `nil`.  This keeps the responsibility for checking for nils and for handling formats in this class and not spread out among other classes too much.  Ex: if a Company does not have an address, the  `Company` class does not have to know how many attributes the exported address string is supposed to have.  Same with a `MembershipApplication` -- they both can rely on the AddressExporter to handle that. 
(In the real world, it's unlikely that a Company will not have an address, but the point remains.)

    This kind of design decision is always a bit of a gray area. It's hard to completely decouple the responsibility from the classes that have the data. I have left `Company` and `MembershipApplication` with _some_ responsibility for handling address exporting/formatting.  But most of it is now encapsulated in the `AddressExporter` class.  And of course there is coupling between the `AddressExporter` and `Address,` but that is reasonable and models the real world.

2. Added these **address fields output to the CSV line f**or a membership_application:
    - company name  (because you mentioned that might be needed as part of the postal address, Susanna)
    - street address 
    - post_code 
    - city 
    - kommun
    - region
    - country

##   Example output:

    'Kontakt e-post','Förnamn','Efternamn','Medlemsnummer','Status','Företag','Gata','Post nr.','Ort','Kommun','Verksamhetslän','Land'
    leon@spray.se,Jon,Viklund,107,Godkänd,"Sandström-Bergström",Erik Tegels Väg 417,'95667,Karlstad,Kungälv,Norrbotten,SE-Sweden
    rima_mattsson@gmail.com,Charlesetta,Dahlberg,,Behandlas,,,,,,,
    rima_mattsson@gmail.com,Charlesetta,Dahlberg,,Ny,,,,,,,
    rima_mattsson@gmail.com,Charlesetta,Dahlberg,,Redo för granskning,,,,,,,

  

   The order can be changed to whatever is needed.
 

### Assumptions made about the the address information for each membership application entry in the CSV file:

(I tried to be sure to capture this in `address_exporter_spec.rb`):

1. the address for a member_application is the `main_address` for the company, or nil if there is no company for the member_application
  
2. the company name has quotes around it ("") so that if it has any punctuation -- like a comma -- it won't be interpreted as multiple fields

2. **the `post_code` starts with a single quote `'`** so that spreadsheet programs won't automatically convert it to a number; it needs to be a string so that any spaces in it are maintained.  Ex:  If the post code is `957 31` then in the CSV line it will be ".....  '957 31,......"

     _This may or may not cause a problem for other programs or services that import it.  Perhaps an easy solution is to output the post_code twice:  one with the leading single quote, and on without it, so that the information is available in both forms._

3.  outputs **SE-Sweden** for the country if the country == Sverige (otherwise just uses what the county is).   This is what the Swedish Postal Service prefers.

4. Can handle having no kommun and can handle having no region

Ready for review:
@thesuss @patmbolger

### CODE REVIEW

File: app/services/address_exporter.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+      "#{address.street_address},'#{address.post_code},#{address.city},#{kommun_str},#{region_str},#{self.country_postal(address)}"
+    else
+      Array.new(num_items, '').join(',')
+    end
+
+  end
+
+
+  private
```
[the remark this answers is not in the document]
Comment (patmbolger) [reply app/services/address_exporter.rb]: `private` method only applies to instance methods.

Here is a good overview of creating private class methods: http://jakeyesbeck.com/2016/01/24/ruby-private-class-methods/?utm_source=rubyweekly&utm_medium=email

File: spec/controllers/admin_controller_spec.rb [hunk 2]
```diff
@@ -13,23 +13,33 @@
   include Devise::Test::ControllerHelpers
 
   # this will bypass Pundit policy access checks so logging in is not necessary
-  before(:each) { Warden.test_mode! }
+  before(:each) {Warden.test_mode!}
```
Comment (patmbolger) [inline spec/controllers/admin_controller_spec.rb]: I thought our style has been to put a space after/before curly braces used in blocks?

File: spec/models/company_spec.rb [hunk 3]
```diff
@@ -97,37 +101,37 @@
...
+    let(:cat1) {create(:business_category, name: 'cat1')}
+    let(:cat2) {create(:business_category, name: 'cat2')}
+    let(:cat3) {create(:business_category, name: 'cat3')}
 
     let(:m1) do
       create(:membership_application, :accepted, user: employee1,
-             num_categories: 0,
-             company_number: company.company_number)
+             num_categories:                           0,
+             company_number:                           company.company_number)
```
Comment (patmbolger) [inline spec/models/company_spec.rb]: Bunch of tabs in this file .....

**Review by patmbolger — CHANGES_REQUESTED** [review]
Please see comments

File: spec/models/company_spec.rb (hunk 3 above)
Comment (weedySeaDragon) [reply spec/models/company_spec.rb]: dang.  I'm surprised those weren't turned to spaces.  I'll definitely get them (and the other stuff) cleaned up.

File: spec/controllers/admin_controller_spec.rb (hunk 2 above)
Comment (weedySeaDragon) [reply spec/controllers/admin_controller_spec.rb]: I don't think so for inline blocks. (I don't think we've been enforcing it.)  But that is actually the style I prefer so I'll change them.

**Review by patmbolger — APPROVED** [review]

**Pull request merged**